Add missing azure vm automate model

bzwei opened the pull request:
Add new class MiqAeServiceManageIQ_Providers_Azure_CloudManager_Vm
Use constantize method to convert string to class

https://bugzilla.redhat.com/show_bug.cgi?id=1276453

gmcculloug commented on app/models/miq_event.rb:
```diff
@@ -24,7 +24,7 @@ def self.raise_evm_event(target, raw_event, inputs = {}, options = {})
       klass, id = target
-      klass = Object.const_get(klass)
+      klass = klass.constantize
       target = klass.find_by_id(id)
```
You can reduce this to: `target = klass.constantize.find_by(:id => id)` and get rid of the `klass = ` line.

gmcculloug commented:
@bzwei Who is the caller that is passing `target` as an array?  I am thinking it might be from the queue.  Do you know if this code path is covered by any tests?

bzwei commented:
@gmcculloug The caller passing an array is the queue. There is a negative test that covers this code path, but the class name is passed in as symbol not a string. `Symbol` does not have `constantize` method. So I added `.to_s` to do the conversion.
I also added a positive test case.
